c10/core/DeviceType.h

```diff
@@ -1,100 +1,16 @@
 #pragma once
 
-// This is directly synchronized with caffe2/proto/caffe2.proto, but
-// doesn't require me to figure out how to get Protobuf headers into
-// ATen/core (which would require a lot more build system hacking.)
-// If you modify me, keep me synchronized with that file.
-
 #include <c10/macros/Export.h>
 
-#include <cstddef>
-#include <cstdint>
-#include <functional>
+// If you modified DeviceType in caffe2/proto/caffe2.proto, please also sync
+// your changes into torch/headeronly/core/DeviceType.h.
```

Comment:
I'm not sure this is true anymore, DeviceType there looks really out of date https://www.internalfb.com/code/fbsource/fbcode/caffe2/caffe2/proto/caffe2.proto 😆

Reply:
The reason we don't have to handle StableIValue conversions for this one is cuz you can't currently pass DeviceType across the boundary (as we're not using it in any shim APIs today). I don't think DeviceType will be passed through the StableIValue boundary until Device, and then we will have to work out the versioning.